### ISSUE TITLE
[turborepo] custom remote cache config.json case sensitivity changed?

### ISSUE THREAD

**Issue by devdumpling** [vercel/turbo#5145]
### What version of Turborepo are you using?

1.9.9

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

My organization has been using TurboRepo with a custom remote server. We have a `config.json` set at our root that defines the following values (I thought at one point these docs existed but I can't seem to find them anywhere):

```json
{
  "teamId": "our_team",
  "apiUrl": "https://our-url.com"
}
```

Has been working a charm up until recently, when we noticed that we were seeing `remote caching disabled` on our turbo runs. 

After some debugging, we managed to fix it by lowercasing the keys:

```json
{
  "teamid": "...",
  "apiurl": "..."
}
```

I can't seem to find any documentation around this anywhere. Was this changed recently/is there somewhere we can keep an eye on to monitor any future changes to the custom remote caching setup?

Any help or insight would be appreciated! 🙏 

Thanks for everything y'all do.

cc @stevejcox 

### Expected Behavior

Would love visibility or documentation regarding custom remote server setups and/or config/spec changes. 

### To Reproduce

Hard to reproduce our enterprise setup exactly, but here's the gist. Let me know if I can provide more detail or clarity.

1. Create custom remote server using spec: https://github.com/vercel/turbo/blob/main/cli/internal/client/client.go
1. Add `.turbo/config.json` to root of consuming monorepo with TurboRepo setup. Add `apiUrl` and `teamId` cased like this. 
1. Run a turbo command and note the `remote caching disabled`. 
1. Change the casing of in `config.json` to `apiurl` and `teamid`. 
1. Run turbo command again and note `remote caching enabled`

**Comment by gsoltis** [comment]
@NicholasLYang this might be parsing via `serde`?

**Comment by NicholasLYang** [comment]
Fixed with #5209
